### PR TITLE
feat(NODE-6161): Sign Releases

### PR DESCRIPTION
**Pull request by aditi-khare-mongoDB**
### Description
Sign releases, tar ball and .node native files. 

#### What is changing?
Add signatures to releases.

##### Is there new documentation needed for these changes?
Yes, README.md has been updated.

#### What is the motivation for this change?
SSDLC compliance.

### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .github/docker/Dockerfile.glibc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+ARG NODE_BUILD_IMAGE=node:16.20.1-bullseye
+FROM $NODE_BUILD_IMAGE AS build
+
+WORKDIR /kerberos
+COPY . .
+
+RUN npm run install kerberos
```
Comment (nbbeeken) [inline .github/docker/Dockerfile.glibc]: Install kerberos lib according to debian instructions

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+permissions:
+  contents: write
+  pull-requests: write
+  id-token: write
+
+jobs:
+  host_builds:
+    strategy:
+      matrix:
+        os: [macos-11, macos-latest, windows-2019]
```
Comment (nbbeeken) [inline .github/workflows/build.yml]: remove macos-11: https://github.com/mongodb-js/mongodb-client-encryption/pull/21

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+
```
Comment (nbbeeken) [inline .github/workflows/build.yml:23]: npm install (check how we setup in FLE, we may also setup a nodejs version) somewhere here

**Comment by aditi-khare-mongoDB** [comment]
https://github.com/mongodb-js/kerberos/actions/runs/9619287117
^^ example run with `npm publish` commmented out

File: README.md [hunk 4]
```diff
@@ -45,6 +45,49 @@ Now you can install `kerberos` with the following:
...
+
+Using the result of the above command, a `curl` command can return the official npm package for the release.
+
+To verify the integrity of the downloaded package, run the following command:
+```shell
+gpg --verify kerberos-X.Y.Z.tgz.sig kerberos-X.Y.Z.tgz
+```
+
+>[!Note]
+No verification is done when using npm to install the package. To ensure release integrity when using npm, download the tarball manually from the GitHub release, verify the signature, then install the package from the downloaded tarball using npm install mongodb-X.Y.Z.tgz.
```
Comment (durran) [inline README.md:87]:
```suggestion
No verification is done when using npm to install the package. To ensure release integrity when using npm, download the tarball manually from the GitHub release, verify the signature, then install the package from the downloaded tarball using npm install kerberos-X.Y.Z.tgz.
```

**Review by durran — CHANGES_REQUESTED** [review]

**Comment by baileympearson** [comment]
duplicates https://github.com/mongodb-js/kerberos/pull/179

**Pull request closed without merging**